display_information/update-graphics/src/main/java/com/esri/samples/update_graphics/UpdateGraphicsSample.java

```diff
@@ -167,39 +159,68 @@ public void start(Stage stage) {
 
       mapView.setOnMouseClicked(e -> {
         if (e.getButton() == MouseButton.PRIMARY && e.isStillSincePress()) {
+
           // clear any selected graphic
           graphicsOverlay.clearSelection();
 
+          // set the cursor to default
+          mapView.setCursor(Cursor.DEFAULT);
+
           // create a point from location clicked
-          Point2D mapViewPoint = new Point2D(e.getX(), e.getY());
-
-          if (isUpdateLocationActive) {
-            // add new location to selected graphic
-            Point mapPoint = mapView.screenToLocation(mapViewPoint);
-            selectedGraphic.setGeometry(mapPoint);
-            isUpdateLocationActive = false;
-          } else {
-            // identify the graphic that was selected
-            identifyGraphics = mapView.identifyGraphicsOverlayAsync(graphicsOverlay, mapViewPoint, 10, false);
-
-            identifyGraphics.addDoneListener(() -> {
-              try {
-                if (!identifyGraphics.get().getGraphics().isEmpty()) {
-                  // store the selected graphic
-                  selectedGraphic = identifyGraphics.get().getGraphics().get(0);
-                  selectedGraphic.setSelected(true);
-                  String style = ((SimpleMarkerSymbol) selectedGraphic.getSymbol()).getStyle().toString();
-                  symbolBox.getSelectionModel().select(style);
-
-                  enableUI(false);
-                } else {
-                  enableUI(true);
-                }
-              } catch (Exception x) {
-                new Alert(Alert.AlertType.ERROR, "Error identifying clicked graphic").show();
+          mapViewPoint = new Point2D(e.getX(), e.getY());
+
+          // identify graphics on the graphics overlay
+          identifyGraphics = mapView.identifyGraphicsOverlayAsync(graphicsOverlay, mapViewPoint, 10, false);
+
+          identifyGraphics.addDoneListener(() -> {
+            try {
+              if (!identifyGraphics.get().getGraphics().isEmpty()) {
+                // get the first identified graphic
+                selectedGraphic = identifyGraphics.get().getGraphics().get(0);
+                // select the identified graphic
+                selectedGraphic.setSelected(true);
+                // update the drop down box with the identified graphic's current symbol
+                String style = ((SimpleMarkerSymbol) selectedGraphic.getSymbol()).getStyle().toString();
+                symbolBox.getSelectionModel().select(style);
+                // show the UI
+                disableUI(false);
+              } else {
+                disableUI(true);
               }
-            });
-          }
+            } catch (Exception x) {
+              new Alert(Alert.AlertType.ERROR, "Error identifying clicked graphic").show();
+            }
+          });
+        }
+      });
+
+      mapView.setOnMouseDragged(e -> {
```

Comment:
It would be really cool if we could identify any potential graphic at the clicked point, select it, and then allow to drag if the mouse pointer is held down. We might be able to do that by extracting a method e.g. `identifyGraphicAtClickedPoint`, and then allowing to select it. Could be a bit tricky to do it elegantly, so I would suggest don't spend too much time on it if the code starts growing arms and legs!

Reply:
@sclaridge perhaps Jon's suggestion combined with my attempt below might work? Like Jon says though, don't spend too long on it, we have a workable solution from the first suggestion I put up.